cli/formatter/format.go

```diff
@@ -0,0 +1,800 @@
+	}
+	result = result + "|"
+
+	return result
+}
+
+// constructMarkdownEmptyRow constructs empty row in markdown notation.
+func constructMarkdownEmptyRow(totalRowAmount int) string {
+	var result = "| "
+	for i := 1; i < totalRowAmount; i++ {
+		result = result + "| "
+	}
+	result = result + "|"
+
+	return result
+}
+
+// renderSingleColumnTable returns table with single column as string.
+func renderSingleColumnTable(renderBatch []renderNode, opts *Opts) string {
+	t := table.NewWriter()
+	var rows []table.Row
+	rows = append(rows, table.Row{"col1"})
+	for _, node := range renderBatch {
+		switch node.(type) {
+		case float64, float32:
+			rows = append(rows, []interface{}{strconv.FormatFloat(node.(float64), 'f', -1, 64)})
+		case nil:
+			rows = append(rows, []interface{}{"nil"})
+		default:
+			rows = append(rows, []interface{}{fmt.Sprintf("%v", node)})
+		}
+	}
+
+	var rowsAmount = len(rows)
+	if opts.TransposeMode {
+		rows = transposeRows(rows)
+	}
+	t.AppendRows(rows)
+	t.Style().Options.SeparateRows = true
+
+	if opts.NoGraphics {
+		t.SetStyle(table.Style{Box: StyleWithoutGraphics})
+	}
+	if opts.ColWidthMax > 0 {
+		handleColWidth(rowsAmount, t, opts)
+	}
+
+	if opts.TableDialect == MarkdownTableDialect {
+		markdownTable := strings.Split(t.RenderMarkdown(), "\n")
+		res := constructMarkdownTable(markdownTable, 1, rowsAmount, opts)
+		return res + "\n"
+	}
+	if opts.TableDialect == JiraTableDialect {
+		return t.RenderMarkdown() + "\n\n"
+	}
+
+	return t.Render() + "\n"
+}
+
+// castRenderNodeToMap casts renderNode to map[string]renderNode.
+func castRenderNodeToMap(node renderNode) (map[string]renderNode, error) {
+	var castedRes map[string]renderNode
+	var err error
+	nodeYaml, err := yaml.Marshal(node)
+	if err != nil {
+		return nil, err
+	}
+	castedRes, err = decodeYamlMap(string(nodeYaml))
+	if err != nil {
+		return nil, err
+	}
+
+	return castedRes, nil
+}
+
+// renderArrs returns table as string for arrays case.
+func renderArrs(renderBatch []renderNode, opts *Opts) (string, error) {
+	t := table.NewWriter()
+
+	var rows []table.Row
+	rows = append(rows, constructHeaderRowForArrs(renderBatch))
+	for _, node := range renderBatch {
+		nodeBytes, err := yaml.Marshal(node)
+		if err != nil {
+			return "", err
+		}
+		nodeIterable, err := decodeYamlArr(string(nodeBytes))
+		if err != nil {
+			return "", err
+		}
+		var row []interface{}
+		for _, rowColVal := range nodeIterable {
+			if getRenderNodeType(rowColVal) == scalarRenderNodeType {
+				switch rowColVal.(type) {
+				case float64, float32:
+					row = append(row, strconv.FormatFloat(rowColVal.(float64), 'f', -1, 64))
+				case nil:
+					row = append(row, "nil")
+				default:
+					row = append(row, fmt.Sprintf("%v", rowColVal))
+				}
+			} else {
+				// Encoding as json value for map/array table cell cases.
+				jsonVal, err := encodeToCellJsonVal(rowColVal)
+				if err != nil {
+					return "", err
+				}
+				row = append(row, jsonVal)
+			}
+		}
```

Comment:
The duplication is still be here.

Reply:
Done, moved to `encodeScalarNodeToTableRow`